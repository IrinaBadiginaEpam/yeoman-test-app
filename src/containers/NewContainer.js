import React from 'react';

export function NewContainer() {
  return (
    <React.Fragment>
      Hey you! This is the new page!
      <a href="/">Go back to the counter</a>
    </React.Fragment>
  );
}

export default NewContainer;
