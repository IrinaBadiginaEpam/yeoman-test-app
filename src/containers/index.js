export { default as CounterContainer } from './CounterContainer'
export { default as NewContainer } from './NewContainer';
