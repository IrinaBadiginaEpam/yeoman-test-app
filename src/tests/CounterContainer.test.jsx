import React from 'react';
import ReactDOM from 'react-dom';
import { CounterContainer } from '../containers/CounterContainer';

it('renders without crashing', () => {
  const div = document.createElement('div'),
  props = {
    increment: () => {},
    decrement: () => {},
    incrementIfOdd: () => {},
    counter: 0
  };
  ReactDOM.render(<CounterContainer {...props}/>, div);
});
