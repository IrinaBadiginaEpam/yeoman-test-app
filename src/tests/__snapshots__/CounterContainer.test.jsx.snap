// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CounterContainer
    counter={0}
    decrement={[Function]}
    increment={[Function]}
    incrementIfOdd={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <pure(Counter)
          counter={0}
          decrement={[Function]}
          increment={[Function]}
          incrementIfOdd={[Function]}
/>,
        <a
          href="/newPage"
>
          Go to the new page
</a>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "counter": 0,
          "decrement": [Function],
          "increment": [Function],
          "incrementIfOdd": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Go to the new page",
          "href": "/newPage",
        },
        "ref": null,
        "rendered": "Go to the new page",
        "type": "a",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <pure(Counter)
            counter={0}
            decrement={[Function]}
            increment={[Function]}
            incrementIfOdd={[Function]}
/>,
          <a
            href="/newPage"
>
            Go to the new page
</a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "counter": 0,
            "decrement": [Function],
            "increment": [Function],
            "incrementIfOdd": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Go to the new page",
            "href": "/newPage",
          },
          "ref": null,
          "rendered": "Go to the new page",
          "type": "a",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CounterContainer
    counter={0}
    decrement={[Function]}
    increment={[Function]}
    incrementIfOdd={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <pure(Counter)
          counter={0}
          decrement={[Function]}
          increment={[Function]}
          incrementIfOdd={[Function]}
/>,
        <a
          href="/newPage"
>
          Go to the new page
</a>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "counter": 0,
          "decrement": [Function],
          "increment": [Function],
          "incrementIfOdd": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Go to the new page",
          "href": "/newPage",
        },
        "ref": null,
        "rendered": "Go to the new page",
        "type": "a",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <pure(Counter)
            counter={0}
            decrement={[Function]}
            increment={[Function]}
            incrementIfOdd={[Function]}
/>,
          <a
            href="/newPage"
>
            Go to the new page
</a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "counter": 0,
            "decrement": [Function],
            "increment": [Function],
            "incrementIfOdd": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Go to the new page",
            "href": "/newPage",
          },
          "ref": null,
          "rendered": "Go to the new page",
          "type": "a",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
